connman/src/wispr.c

```diff
@@ -628,13 +628,15 @@ static void wispr_portal_browser_reply_cb(struct connman_service *service,
 		return;
 
 	if (!authentication_done) {
-		wispr_portal_error(wp_context);
 		free_wispr_routes(wp_context);
+		wispr_portal_error(wp_context);
+		wispr_portal_context_unref(wp_context);
 		return;
 	}
 
 	/* Restarting the test */
 	__connman_wispr_start(service, wp_context->type);
+	wispr_portal_context_unref(wp_context);
```

Comment:
I'm assuming this is because when wispr is restarted it is assumed here to reference the context. But apparently it does manipulate, or create the context. I've no idea if this is correct or not...

Reply:
Aha, I need to hunt down the commit that removed that `|| !authentication_done` from the first checks. But the `!service` case is valid because the context should have been freed at that point if I understand correctly.

Without portal or handling invalid context, well, I guess that is just something that should not really be, error recovery is not easy here. I think that according to the code changing context IP family type also means that the unreffing has been completed for the previous and we shouldn't handle it here now. 

I'll do the modifications to the PR.

Reply:
Pfft, we've added that check in a git history far, far away the other side sources in 81e7f8301b2dce5f8fe1e3ba2f8f7f50201892d3

Reply:
And 653b5b030d0fa34dd7b75b58af073e8dd267307c "make better", that is descriptive.

Reply:
I think the way it now is would have less issues (?).

Reply:
Yep, now this is how upstream also has it. Marking as resolved. If we get any notices about changed behavior on captive portals we should look here for the cause.